common/util.go

```diff
@@ -387,3 +392,8 @@ func JoinURL(url, path string) string {
 	}
 	return url + path
 }
+
+func ReadAtMost(r io.Reader, n int64) ([]byte, error) {
```

Comment:
Not logging any warnings or returning an error here for the case when the segment might have more data than `MaxSegSize` passed in `n` here.

We could do that using manual `Read` calls and increasing buffer gradually if needed - like done in [`ReadAll`'s implementation](https://golang.org/src/io/io.go?s=19249:19287#L616)

Reply:
I think we should return error from here if not all data was read - because if broadcaster didn't downloaded all segment it doesn't make sense to send it to transcoding - anyway it will be broken segment.

Reply:
Makes sense on it being a broken segment and needing error log of some sort, will update!

NB: I was able to transcode partial segment on CPU (haven't tested on GPU), so was uncertain if we should still make an attempt at transcoding or not - but for now makes sense to just error out and reconsider later if we see huge segments.

Reply:
I don't think users expect they segments to be _partially_ transcoded 😄